### PR TITLE
[Small Feature] Add collision detection in PMP

### LINKED ISSUE #3233: AABB_traits does not respect the AABBTraits Concept
## Issue Details
- [x] The AABB_traits class uses a class Sort_primitives instead of implementing the three functors 
Split_primitives_along_(x/y/z)_axis. 

- [x] It is also missing the type Intersect, which is called Intersection in the class.
- [x] All functors should be const but it is not documented.

### PR DESCRIPTION
**Pull request by sloriot**
## Summary of Changes
Add class `Rigid_triangle_mesh_collision_detection` based on the intersection of transformed AABB-trees do make some intersection test with moved triangle meshes

The PR also incorporate some bug fixes in AABB-tree detected while developing this feature.

## Release Management

* Affected package(s): PMP
* fix #3233 
* Link to compiled documentation (obligatory for small feature) [link](https://cgal.geometryfactory.com/~sloriot/doc/CollDetect-v1/Polygon_mesh_processing/classCGAL_1_1Rigid__triangle__mesh__collision__detection.html)
* License and copyright ownership: GF
* Small feature page: [here](https://cgal.geometryfactory.com/CGAL/Members/wiki/Features/Small_Features/PMPCollisionDetection)
* **Pre-approved** -- @afabri 16:04, 4 October 2018 (CEST) 

- [x] TODO: update `CHANGES.md`

### CODE REVIEW

File: Polygon_mesh_processing/include/CGAL/Rigid_triangle_mesh_collision_detection.h [hunk 1]
```diff
@@ -49,26 +49,26 @@ namespace CGAL {
...
  *                      or if only translations and scalings are applied (`Tag_false`). Some optimizations are
  *                      switch on in case there are no rotations.
  */
 template <class TriangleMesh,
           class VertexPointMap = Default,
-          class KernelType = Default,
-          class AABBTreeType = Default,
+          class Kernel = Default,
+          class AABBTree = Default,
           class HAS_ROTATION = CGAL::Tag_true>
```
Comment (lrineau) [inline Polygon_mesh_processing/include/CGAL/Rigid_triangle_mesh_collision_detection.h]: `HAS_ROTATION` does not respect our naming guidelines. I should be `Has_rotation` or `Has_rotation_tag`.

**Comment by lrineau** [comment]
[Compilation error in Travis](https://travis-ci.org/CGAL/cgal/jobs/437155396#L2152):
```
[  2%] Building CXX object CMakeFiles/remeshing_test_P_SM_OM.dir/remeshing_test_P_SM_OM.cpp.o
/home/travis/build/CGAL/cgal/Polygon_mesh_processing/test/Polygon_mesh_processing/test_pmp_collision_detection.cpp:96:5: error: 
      no matching constructor for initialization of 'typename
      CGAL::Rigid_triangle_mesh_collision_detection<Surface_mesh<Point_3<Epick>
      > >::AABB_tree' (aka 'CGAL::AABB_tree<CGAL::AABB_traits<CGAL::Epick,
      CGAL::AABB_face_graph_triangle_primitive<CGAL::Surface_mesh<CGAL::Point_3<CGAL::Epick>
      >, CGAL::Surface_mesh<CGAL::Point_3<CGAL::Epick>
      >::Property_map<CGAL::SM_Vertex_index, CGAL::Point_3<CGAL::Epick> >,
      CGAL::Boolean_tag<true>, CGAL::Boolean_tag<false> >, CGAL::Default> >')
    tm1_tree(boost::begin(faces(tm1)), boost::end(faces(tm1)), tm1, get(boost::vertex_point, tm1));
    ^        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

**Comment by sloriot** [comment]
I can reproduce when switching off c++11.

**Comment by lrineau** [comment]
There is now a conflict with `master`.

**Comment by lrineau** [comment]
[One platform signals this warning][*]:
```
test/Polygon_mesh_processing/test_pmp_collision_detection.cpp:174:2: warning: no newline at end of file
```

[*]: https://cgal.geometryfactory.com/CGAL/testsuite/CGAL-4.14-Ic-19/Polygon_mesh_processing/TestReport_lrineau_CentOS5.gz

**Comment by sloriot** [comment]
Successfully tested in [4.14-Ic-30](https://cgal.geometryfactory.com/CGAL/testsuite/results-4.14-Ic-30.shtml).

**Pull request merged**